google/cloud/aiplatform/featurestore/featurestore.py

```diff
@@ -301,24 +302,253 @@ def list_entity_types(
+            force (bool):
+                Optional. If force is set to True, all features in each entityType
+                will be deleted prior to entityType deletion. Default is False.
             sync (bool):
                 Optional. Whether to execute this deletion synchronously. If False, this method
                 will be executed in concurrent Future and any downstream object will
                 be immediately returned and synced when the Future has completed.
         """
-        entity_types = []
-        for entity_type_id in entity_type_ids:
-            entity_type = self.get_entity_type(entity_type_id=entity_type_id)
-            entity_type.delete(sync=False)
-            entity_types.append(entity_type)
+        if not entity_type_ids:
+            entity_types = self.list_entity_types()
+        elif entity_type_ids and isinstance(entity_type_ids, list):
+            entity_types = [
+                self.get_entity_type(entity_type_id=entity_type_id)
+                for entity_type_id in entity_type_ids
+            ]
+
+        for entity_type in entity_types:
+            entity_type.delete(force=force, sync=False)
 
         for entity_type in entity_types:
             entity_type.wait()
+
+    @base.optional_sync()
+    def delete(self, force: bool = None, sync: bool = True) -> None:
+        """Deletes this Featurestore resource. If force is set to True,
+        all entityTypes in this Featurestore will be deleted prior to featurestore deletion,
+        and all features in each entityType will be deleted prior to each entityType deletion.
+
+        WARNING: This deletion is permanent.
+
+        Args:
+            force (bool):
+                If set to true, any EntityTypes and
+                Features for this Featurestore will also
+                be deleted. (Otherwise, the request will
+                only work if the Featurestore has no
+                EntityTypes.)
+            sync (bool):
+                Whether to execute this deletion synchronously. If False, this method
+                will be executed in concurrent Future and any downstream object will
+                be immediately returned and synced when the Future has completed.
+        Raises:
+            FailedPrecondition: If entityTypes are created in this Featurestore and force = False.
+        """
+        _LOGGER.log_action_start_against_resource("Deleting", "", self)
+        lro = getattr(self.api_client, self._delete_method)(
+            name=self.resource_name, force=force
+        )
+        _LOGGER.log_action_started_against_resource_with_lro(
+            "Delete", "", self.__class__, lro
+        )
+        lro.result()
+        _LOGGER.log_action_completed_against_resource("deleted.", "", self)
+
+    @classmethod
+    @base.optional_sync()
+    def create(
+        cls,
+        featurestore_id: str,
+        online_store_fixed_node_count: Optional[int] = 1,
```

Comment:
does the Optional mean it can be omit in the function call (and take the default value of 1), or it can be None? If the later, then we probably remove the Optional (but keep defaut value)?

Reply:
For the current setting, the `online_store_fixed_node_count` should set to `Required` with 1 as default. But since there are some pending updates i.e. no online store launched by default when creating fs, I make it Optional to None, but in docstring, indicated when not specified 1 is in use.